Sources/SmileID/Classes/SelfieCapture/SelfieViewModel.swift

```diff
@@ -314,20 +305,59 @@ public class SelfieViewModel: ObservableObject, ARKitSmileDelegate {
                     )
                 }
                 let authResponse = try await SmileID.api.authenticate(request: authRequest).async()
-                let prepUploadRequest = PrepUploadRequest(
-                    partnerParams: authResponse.partnerParams.copy(extras: extraPartnerParams),
-                    allowNewEnroll: String(allowNewEnroll),
-                    timestamp: authResponse.timestamp,
-                    signature: authResponse.signature
-                )
-                let prepUploadResponse = try await SmileID.api.prepUpload(
-                    request: prepUploadRequest
-                ).async()
-                _ = try await SmileID.api.upload(
-                    zip: zip,
-                    to: prepUploadResponse.uploadUrl
-                ).async()
-                didSubmitSmartSelfieJob = true
+
+                var smartSelfieLivenessImages = [MultipartBody]()
+                var smartSelfieImage: MultipartBody?
+                if let selfie = try? Data(contentsOf: selfieImage), let media = MultipartBody(
+                    withImage: selfie,
+                    forKey: selfieImage.lastPathComponent,
+                    forName: selfieImage.lastPathComponent
+                ) {
+                    smartSelfieImage = media
+                }
+                if !livenessImages.isEmpty {
+                    let livenessImageInfos = livenessImages.compactMap { liveness -> MultipartBody? in
+                        if let data = try? Data(contentsOf: liveness) {
+                            return MultipartBody(
+                                withImage: data,
+                                forKey: liveness.lastPathComponent,
+                                forName: liveness.lastPathComponent
+                            )
+                        }
+                        return nil
+                    }
+
+                    smartSelfieLivenessImages.append(contentsOf: livenessImageInfos.compactMap { $0 })
+                }
+                guard let smartSelfieImage = smartSelfieImage,
+                smartSelfieLivenessImages.count == numLivenessImages else {
+                    throw SmileIDError.unknown("Selfie capture failed")
+                }
+                let response = if allowNewEnroll {
```

Comment:
Auth vs Enroll is controlled by `isEnroll`, not `allowNewEnroll`

```suggestion
                let response = if isEnroll {
```

Reply:
Good catch, fixed